Assets/MixedRealityToolkit.Services/CameraSystem/MixedRealityCameraSystem.cs

```diff
@@ -52,61 +44,137 @@ public bool IsOpaque
         /// <inheritdoc />
         public string SourceName { get; } = "Mixed Reality Camera System";
 
-        private MixedRealityCameraProfile cameraProfile = null;
-
         /// <inheritdoc/>
-        public MixedRealityCameraProfile CameraProfile
+        public MixedRealityCameraProfile CameraProfile =>  ConfigurationProfile as MixedRealityCameraProfile;
+
+        private DisplayType currentDisplayType;
+        private bool cameraOpaqueLastFrame = false;
+
+        /// <summary>
+        /// Specifies whether or not the camera system was able to register a camera settings provider.
+        /// If so, camera management is not to be performed by the camera system itself. If not, the camera
+        /// system is to behave as it did in MRTK versions 2.0.0 and 2.1.0.
+        /// </summary>
+        private bool useFallbackBehavior = true;
+
+        /// <inheritdoc />
+        public override void Initialize()
         {
-            get
+            MixedRealityCameraProfile profile = ConfigurationProfile as MixedRealityCameraProfile;
+
+            if ((GetDataProviders<IMixedRealityCameraSettingsProvider>().Count == 0) && (profile != null))
             {
-                if (cameraProfile == null)
+                // Register the settings providers.
+                for (int i = 0; i < profile.SettingsConfigurations.Length; i++)
                 {
-                    cameraProfile = ConfigurationProfile as MixedRealityCameraProfile;
+                    MixedRealityCameraSettingsConfiguration configuration = profile.SettingsConfigurations[i];
+
+                    if (configuration.ComponentType?.Type == null) 
+                    { 
+                        // Incomplete configuration, do not try to register until a type is set in the profile.
+                        continue; 
+                    }
+
+                    object[] args = { Registrar, this, configuration.ComponentName, configuration.Priority, configuration.SettingsProfile };
+
+                    if (RegisterDataProvider<IMixedRealityCameraSettingsProvider>(
```

Comment:
Should we consider making common method of this in each input system, spatial awareness, and now here?

Reply:
we should co-engineer a solution that also confirms whether or not the provider successfully initialized. i can go in more detail in a separate discussion / PR

Reply:
#6372